vortex-layout/src/layouts/struct_/writer.rs

```
@@ -63,6 +63,7 @@ impl LayoutWriter for StructLayoutWriter {
         self.row_count += struct_array.len() as u64;
 
         for i in 0..struct_array.nfields() {
+            // TODO(joe): handle struct validity
```

Could you at least return an error if the struct is nullable and isn't AllValid?